### DISCUSSION TITLE
An example on heteroskedastic noise with StudentT likelihood

### DISCUSSION THREAD

**Discussion by IvanDeBoi** [GPflow/GPflow#1730] [Q&A] [answered]
*Is there a feature that needs some example code in a notebook?*

Hi,
I would like a notbook like this:
https://gpflow.readthedocs.io/en/master/notebooks/advanced/heteroskedastic.html

but with Student-t likelihood as described in http://proceedings.mlr.press/v51/saul16.html.

I can not just change this:

`likelihood = gpf.likelihoods.HeteroskedasticTFPConditional(
    distribution_class=tfp.distributions.Normal,  # Gaussian Likelihood
    scale_transform=tfp.bijectors.Exp(),  # Exponential Transform
)`

into this:

`likelihood = gpf.likelihoods.HeteroskedasticTFPConditional(
    distribution_class=tfp.distributions.StudentT
    scale_transform=tfp.bijectors.Exp(),  # Exponential Transform
)`

because then I get the following error in the optimization:
missing 1 required positional argument: 'scale'

An example on this would be helpful.

Kind regards,

Ivan

**Comment by gustavocmv** [comment, marked as answer]
This happens because `HeteroskedasticTFPConditional` calls the `distribution_class` as `distribution_class(loc, scale)`, using positional arguments. However, `tfp.distributions.StudentT` expects signature is `tfp.distributions.StudentT(df, loc, scale)`.

There are some possibilities to work around it. If you want a fixed value for the `df` parameter, you can just call:
```
df = 3.0
likelihood = gpf.likelihoods.HeteroskedasticTFPConditional(
    distribution_class=lambda loc, scale: tfp.distributions.StudentT(df=df, loc, scale),
    scale_transform=tfp.bijectors.Exp(), # Exponential Transform
)
```

If you want to have the `df` parameter not fixed, I'd suggest subclassing `MultiLatentTFPConditional`.

**Comment by IvanDeBoi** [comment]
Great, I thank you very much!
